streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -231,6 +240,69 @@ private void restoreTasks(final long now) {
             }
         }
 
+        private void maybeGetClientInstanceIds() {
+            if (fetchDeadlineClientInstanceId != -1) {
+                if (!clientInstanceIdFuture.isDone()) {
+                    if (fetchDeadlineClientInstanceId >= time.milliseconds()) {
+                        try {
+                            // if the state-updated thread has active work:
+                            //    we pass in a timeout of zero into each `clientInstanceId()` call
+                            //    to just trigger the "get instance id" background RPC;
+                            //    we don't want to block the state updater thread that can do useful work in the meantime
+                            // otherwise, we pass in 100ms to avoid busy waiting
+                            clientInstanceIdFuture.complete(
+                                restoreConsumer.clientInstanceId(
+                                    allWorkDone() ? Duration.ofMillis(100L) : Duration.ZERO
```

Comment:
I think this will always be false here, because of `fetchDeadlineClientInstanceId`

Reply:
Good catch!